ShaderTool/Command/Map.cs

```diff
@@ -486,30 +497,32 @@ class Map {
                 uint actorDataSize = (uint)actorData.vertices.Length * 4;
                 StreamWrite(actorDataSize);
 
-                if ((actorData.localTransform.Length - 16) % 4 != 0) {
-                    Console.WriteLine("Wrong transform size, needs to be a multiple of 4");
-                }
+                if (actorData.localTransform.Length != 16)
+                    Console.WriteLine("Wrong transform size, needs to be 16 is {0}", actorData.localTransform.Length);
 
-                uint matrixCount = (uint)Math.Truncate((actorData.localTransform.Length - 16) / 4.0f);
-                if (matrixCount == 1) {
-                    Console.WriteLine("Possibly wrong instance count, lower than 2. Make sure to include your basic matrix for all objects and the others after the basic matrix.");
-                }
-                StreamWrite(matrixCount);
+                for (int i = 0; i < 16; i++)
+                    StreamWrite(actorData.localTransform[i]);
 
-                // Write the local transform as a 4x4 matrix into the file
-                for (int y = 0; y < (16 + matrixCount * 4); y++)
-                    StreamWrite(actorData.localTransform[y]);
+                StreamWrite(0); // TODO Animation ID
 
-                // Find the material ID from the material name
-                byte id = (byte)Array.IndexOf(mapData.materialNames, actorData.materialName);
+                // List is always a multiple of 4
+                int actorTransformID = transformList.Count / 4;
+                StreamWrite(actorTransformID);
 
                 // Just take the last material in case none was supplied
                 // We can avoid excessive error handling that way
-                StreamWrite(id);
+                int materialindex = Array.IndexOf(mapData.materialNames, actorData.materialName);
```

Comment:
why was this changed to an int? isnt the limit still 256?

Reply:
This has been changed primarily for alignment purposes.